src/thirdparty/axom/fmt/format-inl.h

```diff
@@ -121,11 +121,17 @@ template <typename Char> AXOM_FMT_FUNC Char decimal_point_impl(locale_ref) {
 AXOM_FMT_API AXOM_FMT_FUNC format_error::~format_error() noexcept = default;
 #endif
 
-AXOM_FMT_FUNC std::system_error vsystem_error(int error_code, string_view format_str,
+// BEGIN AXOM BUGFIX
+// NVCC's preprocessor converts 'std::error_code' to `class std::error_code'
+// and then complains that `return class std::error_code' isn't valid!
+using axom_fmt_system_error = std::system_error;
+
+AXOM_FMT_FUNC axom_fmt_system_error vsystem_error(int error_code, string_view format_str,
                                          format_args args) {
   auto ec = std::error_code(error_code, std::generic_category());
-  return std::system_error(ec, vformat(format_str, args));
+  return axom_fmt_system_error(ec, vformat(format_str, args));
```

Comment:
As far as I can tell, `nvcc`'s preprocessor appears to be replacing `std::runtime_error` with `class std::runtime_error` when the code has `using namespace std`

The errors here and below were:

```
<axom>/src/thirdparty/axom/fmt/format-inl.h:127:8: error: expected expression
return class std::system_error(ec, vformat(format_str, args)); 
       ^
<axom>/src/thirdparty/axom/fmt/format-inl.h:1456:32: error: expected expression
write(std::back_inserter(out), class std::system_error(ec, message).what()); 
                               ^
```
Using a type alias circumvents this problem.

Reply:
I'll try to create a simple reproducer and log an issue.